src/components/inputs/time-input/time-input.js

```diff
@@ -1,354 +1,187 @@
+  if (seconds === 0 && milliseconds === 0) return base;
+  if (milliseconds === 0) return `${base}:${leftPad(seconds)}`;
+  // string representation of a time without timezone in ISO 8601 format
+  return `${base}:${leftPad(seconds)}.${rightPad(milliseconds)}`;
 };
 
-// Converts Date object provided by Flatpickr to formats expected by Datepicker users
-export const presentOutput = ({ value, timeScale, timeZone }) => {
-  switch (timeScale) {
-    case 'time':
-      return moment(value).format('HH:mm:ss.SSS');
-    case 'datetime': {
-      // As we shifted datetime value before passing it to Flatpickr, now we have to
-      // shift it back
-      return moment(value)
-        .subtract(getFlatpickrOffset(value, timeZone), 'minutes')
-        .toISOString();
-    }
-    case 'date':
-      return moment(value).format('YYYY-MM-DD');
-    default:
-      return value;
-  }
-};
-
-export const createFormatter = (timeScale, locale) => value => {
-  switch (timeScale) {
-    case 'time':
-      return moment(value, 'HH:mm:ss.SSS')
-        .locale(locale)
-        .format('LT');
-    case 'datetime':
-      return moment(value)
-        .locale(locale)
-        .format('L LT');
-    case 'date':
-      return moment(value)
-        .locale(locale)
-        .format('L');
-    default:
-      return value;
-  }
+// Attempts to parse a string containing a time in either 12h or 24h format,
+// with precision of up to three milliseconds
+// Valid inputs:
+//   13:00
+//   3:00
+//   3 PM
+//   14:5 am
+//   13:00:00.000
+//   13:00:60
+//   13:00:59.908
+// Returns an array containing
+//   [hours, minutes, seconds, milliseconds]
+// or null
+const parseTime = rawTime => {
+  if (!rawTime || typeof rawTime !== 'string') return null;
+
+  const time = rawTime.trim().toLowerCase();
+
+  const match = time.match(
+    /^(\d{1,2})(?::(\d{1,2})(?::(\d{1,2})(?:\.(\d{1,3}))?)?)?\s*(am|pm)?$/
+  );
+  if (!match) return null;
+
+  // As we accept eg "3 AM" there might not be a value for minutes, seconds or
+  // milliseconds, so we default them
+  const [
+    ,
+    hours,
+    minutes = '00',
+    seconds = '00',
+    milliseconds = '000',
+    amPm,
+  ] = match;
+  if (Number(minutes) > 59) return null;
+  if (amPm && Number(hours) > 12) return null;
+  if (!amPm && Number(hours) > 23) return null;
+  if (Number(seconds) > 59) return null;
+  if (Number(milliseconds) > 999) return null;
+
+  return {
+    hours: Number(hours) + (amPm === 'pm' ? 12 : 0),
+    minutes: Number(minutes),
+    seconds: Number(seconds),
+    milliseconds: Number(milliseconds),
+    hasSeconds: Number(seconds) !== 0 || Number(milliseconds) !== 0,
+    hasMilliseconds: Number(milliseconds) !== 0,
+    amPm,
+  };
 };
 
-export class TimeInput extends React.PureComponent {
+// This component lets the user select a time.
+//
+// The `value` of this component and the value of the event `onChange` gets
+// called with is always a 24hr format time string, or an empty string.
```

Comment:
This is outdated, the value can be any string in any format. Only `TimeInput.to24h(value)` will return the 24h format or an empty string. 

The comment needs to be adjusted.